ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -341,6 +346,46 @@ let hover (state : State.t) { HoverParams.textDocument = { uri }; position } =
     let resp = Hover.create ~contents ~range () in
     Ok (Some resp, state)
 
+let signature_help (state : State.t)
+    { SignatureHelpParams.textDocument = { uri }; position; context = _ } =
+  let store = state.store in
+  let uri = Uri.t_of_yojson (`String uri) in
+  let open Fiber.Result.O in
+  let* doc = Fiber.return (Document_store.get store uri) in
+  let pos = Position.logical position in
+  let open Fiber.O in
+  let+ result =
+    Document.with_pipeline doc (fun pipeline ->
+        let typer = Mpipeline.typer_result pipeline in
+        let pos = Mpipeline.get_lexing_pos pipeline pos in
+        let node = Mtyper.node_at typer pos in
+        let context = Merlin_analysis.Completion.application_signature node in
+        match context with
+        | `Application { fun_name; signature; param_offset } ->
+          let fun_name = Option.value ~default:"_" fun_name in
+          let prefix = sprintf " %s : " fun_name in
+          let offset = String.length prefix in
+          let parameters =
+            match param_offset with
+            | Some (param_start, param_end) ->
+              [ ParameterInformation.create
```

Comment:
How comes there's always only one parameter? Shouldn't we show all of them here and pick the right one with `activeParameter`?